http/service.go

```diff
@@ -1091,7 +1096,17 @@ func (s *Service) execute(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, ErrLeaderNotFound.Error(), http.StatusServiceUnavailable)
 			return
 		}
-		results, resultsErr = s.cluster.Execute(er, addr, timeout)
+
+		username, password, ok := r.BasicAuth()
+		if !ok {
+			username = ""
+		}
+
+		results, resultsErr = s.cluster.Execute(er, addr, username, password, timeout)
+		if resultsErr != nil && resultsErr.Error() == "Unauthorized" {
```

Comment:
Checking if `Error` is equal to a specific string is obviously brittle. I wonder if we should add a dedicated flag to the `resultsErr` struct, indicating "no authorization". WDYT?

Reply:
Agreed, sort of anticipated this feedback :).

Reply:
Added a lot more structure around the errors returned from the TCP service.  Let me know what you think/if you may prefer to take a different approach.